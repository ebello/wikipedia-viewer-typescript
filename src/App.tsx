/** @jsxRuntime classic */
/** @jsx jsx */
import { jsx, Styled, Container } from 'theme-ui';
import PropTypes from 'prop-types';
import { Router, RouteComponentProps } from '@reach/router';
import { Global } from '@emotion/core';
import WikipediaViewer from './contexts/wikipedia-viewer';
import SearchPages from './components/search-pages';
import PageViewer from './components/page-viewer';

interface WikiProps extends RouteComponentProps {
  pageTitle?: string,
}

const WikiPage = ({ pageTitle }: WikiProps) => (
  <WikipediaViewer>
    <Styled.root>
      <Global
        styles={{
          html: {
            boxSizing: 'border-box',
          },
          '*, *:before, *:after': {
            boxSizing: 'inherit',
          },
          body: {
            margin: 0,
          },
        }}
      />
      <Container sx={{ pt: [0, 0] }}>
        <header
          sx={{
            position: 'sticky',
            top: 0,
            bg: 'background',
            py: [3, 4],
          }}
        >
          <SearchPages />
        </header>
        <main>
          {pageTitle && <PageViewer title={pageTitle} />}
        </main>
      </Container>
    </Styled.root>
  </WikipediaViewer>
);

WikiPage.propTypes = {
  pageTitle: PropTypes.string,
};
WikiPage.defaultProps = {
  pageTitle: '',
};

const Wiki = (): JSX.Element => (
  <Router>
    <WikiPage path="/" />
    <WikiPage path=":pageTitle" />
  </Router>
);

export default Wiki;
