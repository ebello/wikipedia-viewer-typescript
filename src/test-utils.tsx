import React, { ReactNode } from 'react';
import { render, RenderResult, RenderOptions } from '@testing-library/react';
import {
  createHistory,
  createMemorySource,
  LocationProvider,
  History,
} from '@reach/router';
import { ThemeProvider } from 'theme-ui';
import theme from './theme';

// const AllTheProviders = ({ children }: { children: ReactNode }): React.FC<{ children: ReactNode }> => (
//   <ThemeProvider theme={theme}>
//     {children}
//   </ThemeProvider>
// );

type ProviderProps = {
  children: ReactNode,
};

const AllTheProviders: React.FunctionComponent<ProviderProps> = ({ children }: ProviderProps) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
);

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: AllTheProviders, ...options });

export const renderWithRouter = (
  ui: React.ReactElement,
  { route = '/', history = createHistory(createMemorySource(route)) } = {},
): RenderResult & { history: History } => ({
  ...render(<LocationProvider history={history}>{ui}</LocationProvider>),
  // adding `history` to the returned utilities to allow us
  // to reference it in our tests (just try to avoid using
  // this to test implementation details).
  history,
});

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };

export const sleep = (ms: number): Promise<unknown> => new Promise(
  (resolve) => setTimeout(resolve, ms),
);
