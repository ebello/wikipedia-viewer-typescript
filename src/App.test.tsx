import React from 'react';
import { renderWithRouter } from './test-utils';
import App, { WikiPage } from './App';

describe('WikiPage', () => {
  it('renders', () => {
    const { asFragment } = renderWithRouter(<WikiPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with a page title', () => {
    const { asFragment } = renderWithRouter(<WikiPage pageTitle="testing" />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('App', () => {
  it('renders home page', () => {
    const { container } = renderWithRouter(<App />);
    expect(container.querySelector('main')).toBeNull();
  });

  it('renders page url', () => {
    const { container } = renderWithRouter(<App />, { route: '/wiki/testing' });
    expect(container.querySelector('main')).not.toBeNull();
  });
});
