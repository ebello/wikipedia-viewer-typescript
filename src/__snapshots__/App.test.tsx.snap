// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WikiPage renders 1`] = `
<DocumentFragment>
  .emotion-4 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;
  line-height: 1.5;
  color: var(--theme-ui-colors-text,#232323);
  background-color: var(--theme-ui-colors-background,#fff);
  font-size: 14px;
}

@media screen and (min-width:46.25em) {
  .emotion-4 {
    font-size: 16px;
  }
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  max-width: 1400px;
  margin-left: auto;
  margin-right: auto;
  padding-top: 0;
}

@media screen and (min-width:46.25em) {
  .emotion-3 {
    padding-top: 0;
  }
}

.emotion-2 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  background-color: var(--theme-ui-colors-background,#fff);
  padding-top: 16px;
  padding-bottom: 16px;
}

@media screen and (min-width:46.25em) {
  .emotion-2 {
    padding-top: 32px;
    padding-bottom: 32px;
  }
}

.emotion-1 {
  display: block;
  width: 100%;
}

@media screen and (min-width:46.25em) {
  .emotion-1 {
    display: inline-block;
    width: 50%;
  }
}

.emotion-0 {
  width: 100%;
  font-size: 16px;
}

@media screen and (min-width:46.25em) {
  .emotion-0 {
    font-size: 24px;
  }
}

<div
    class="emotion-4"
  >
    <div
      class="emotion-3"
    >
      <header
        class="emotion-2"
      >
        <div
          class="emotion-1"
          data-reach-combobox=""
        >
          <input
            aria-autocomplete="both"
            aria-controls="listbox--1"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Pages"
            class="emotion-0"
            data-reach-combobox-input=""
            placeholder="Search Wikipedia"
            role="combobox"
            value=""
          />
        </div>
      </header>
    </div>
  </div>
</DocumentFragment>
`;

exports[`WikiPage renders with a page title 1`] = `
<DocumentFragment>
  .emotion-4 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;
  line-height: 1.5;
  color: var(--theme-ui-colors-text,#232323);
  background-color: var(--theme-ui-colors-background,#fff);
  font-size: 14px;
}

@media screen and (min-width:46.25em) {
  .emotion-4 {
    font-size: 16px;
  }
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  max-width: 1400px;
  margin-left: auto;
  margin-right: auto;
  padding-top: 0;
}

@media screen and (min-width:46.25em) {
  .emotion-3 {
    padding-top: 0;
  }
}

.emotion-2 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  background-color: var(--theme-ui-colors-background,#fff);
  padding-top: 16px;
  padding-bottom: 16px;
}

@media screen and (min-width:46.25em) {
  .emotion-2 {
    padding-top: 32px;
    padding-bottom: 32px;
  }
}

.emotion-1 {
  display: block;
  width: 100%;
}

@media screen and (min-width:46.25em) {
  .emotion-1 {
    display: inline-block;
    width: 50%;
  }
}

.emotion-0 {
  width: 100%;
  font-size: 16px;
}

@media screen and (min-width:46.25em) {
  .emotion-0 {
    font-size: 24px;
  }
}

<div
    class="emotion-4"
  >
    <div
      class="emotion-3"
    >
      <header
        class="emotion-2"
      >
        <div
          class="emotion-1"
          data-reach-combobox=""
        >
          <input
            aria-autocomplete="both"
            aria-controls="listbox--3"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Pages"
            class="emotion-0"
            data-reach-combobox-input=""
            placeholder="Search Wikipedia"
            role="combobox"
            value=""
          />
        </div>
      </header>
      <main />
    </div>
  </div>
</DocumentFragment>
`;
